both/access_control.js

```diff
@@ -258,38 +257,24 @@ if(Meteor.isServer) {
         'updatedAt',
         'createdAt',
       ];
-      return _onlyAllowedFields(fields, allowedFields);
+      return onlyAllowedFields(fields, allowedFields);
     },
     fetch: [ 'competitionId' ],
   });
 
   Registrations.allow({
     insert: function(userId, registration) {
-      return Registrations.userIsAllowed(userId, registration);
+      return Registrations.allowOperation(userId, registration, []);
```

Comment:
I have good news and bad news.

Good: **Object.keys(registration)** works exactly as you'd want, and the code would check field names as intended.

Bad: This would make it impossible to register. **countryId, gender, dob**, and **checkedInEvents** are required fields in the SimpleSchema, but are not in the allowed list in access_control.js. So inserting a valid Registration is impossible. This code is at war with itself!

Not sure what the best solution is, but I think it's beyond the scope of this PR.